dimod/reference/composites/higherordercomposites.py

```diff
@@ -129,6 +144,48 @@ def sample_poly(self, poly, penalty_strength=1.0,
                                   discard_unsatisfied=discard_unsatisfied)
 
 
+def expand_initial_state(bqm, initial_state):
+    """Determine the values for the initial state for a binary quadratic model
+    generated from a higher order polynomial.
+
+    Args:
+        bqm (:obj:`.BinaryQuadraticModel`): a bqm object that contains
+            its reduction info.
+
+        initial_state (dict):
+            An initial state for the higher order polynomial that generated the
+            binary quadratic model.
+
+    Returns:
+        dict: A fully specified initial state.
+
+    """
+    # Developer note: this function relies heavily on assumptions about the
+    # existance and structure of bqm.info['reduction']. We should consider
+    # changing the way that the reduction information is passed.
```

Comment:
I think in some future version of the BQM we will stop allowing users to add generic  info. But that has a lot of implications and I think this is fine for now

Reply:
I don't see why would the BQM would say no to kind users in need of generic info storing. :smiley_cat:

Reply:
@randomir i sense heavy sarcasm :smirk_cat:

Reply:
@conta877, actually I wasn't being sarcastic at all. I think allowing more flexibility rather than less is the Pythonic thing to do, and the right approach for Ocean.